src/scenario/action_types/action_handler.c

```diff
@@ -0,0 +1,88 @@
+#include "scenario/action_types/rome_wages.h"
+#include "scenario/action_types/savings.h"
+#include "scenario/action_types/send_standard_message.h"
+#include "scenario/action_types/trade_price.h"
+#include "scenario/action_types/trade_problem_land.h"
+#include "scenario/action_types/trade_problem_sea.h"
+#include "scenario/action_types/trade_route_amount.h"
+#include "scenario/action_types/trade_price_set.h"
+
+void scenario_action_type_init(scenario_action_t *action)
+{
+    switch (action->type) {
+        case ACTION_TYPE_ADJUST_CITY_HEALTH:
+            scenario_action_type_city_health_init(action);
+            break;
+        case ACTION_TYPE_ADJUST_MONEY:
+            scenario_action_type_money_add_init(action);
+            break;
+        case ACTION_TYPE_ADJUST_ROME_WAGES:
+            scenario_action_type_rome_wages_init(action);
+            break;
+        case ACTION_TYPE_ADJUST_SAVINGS:
+            scenario_action_type_savings_add_init(action);
+            break;
+        default:
+            break;
+    }
+}
+
+int scenario_action_type_execute(scenario_action_t *action)
+{
+    switch (action->type) {
+        case ACTION_TYPE_ADJUST_CITY_HEALTH: return scenario_action_type_city_health_execute(action);
+        case ACTION_TYPE_ADJUST_FAVOR: return scenario_action_type_favor_add_execute(action);
+        case ACTION_TYPE_ADJUST_ROME_WAGES: return scenario_action_type_rome_wages_execute(action);
+        case ACTION_TYPE_ADJUST_MONEY: return scenario_action_type_money_add_execute(action);
+        case ACTION_TYPE_ADJUST_SAVINGS: return scenario_action_type_savings_add_execute(action);
+        case ACTION_TYPE_CHANGE_ALLOWED_BUILDINGS: return scenario_action_type_change_allowed_buildings_execute(action);
+        case ACTION_TYPE_CHANGE_RESOURCE_PRODUCED: return scenario_action_type_change_resource_produced_execute(action);
+        case ACTION_TYPE_EMPIRE_MAP_CONVERT_FUTURE_TRADE_CITY: return scenario_action_type_empire_map_convert_future_trade_city_execute(action);
+        case ACTION_TYPE_GLADIATOR_REVOLT: return scenario_action_type_gladiator_revolt_execute(action);
+        case ACTION_TYPE_REQUEST_IMMEDIATELY_START: return scenario_action_type_request_immediately_start_execute(action);
+        case ACTION_TYPE_SEND_STANDARD_MESSAGE: return scenario_action_type_send_standard_message_execute(action);
+        case ACTION_TYPE_TRADE_ADJUST_PRICE: return scenario_action_type_trade_price_adjust_execute(action);
+        case ACTION_TYPE_TRADE_ADJUST_ROUTE_AMOUNT: return scenario_action_type_trade_route_amount_execute(action);
+        case ACTION_TYPE_TRADE_PROBLEM_LAND: return scenario_action_type_trade_problems_land_execute(action);
+        case ACTION_TYPE_TRADE_PROBLEM_SEA: return scenario_action_type_trade_problems_sea_execute(action);
+        case ACTION_TYPE_TRADE_SET_PRICE: return scenario_action_type_trade_price_set_execute(action);
+        default:
+            return 0;
+            break;
```

Comment:
Nitpick: you don't really need the `break` after `return`ing.
Also, since the function names are rather large, I think they would look better on their own line instead of being on the `case` line.